### PR TITLE
Added custom FPL support

### PR DESCRIPTION
**Pull request by NikoDervaric**
Scenario Loader now supports spawning aircraft with predefined flightplans.

### CODE REVIEW

File: Plugins/ScenarioLoader/Plugin.cs [hunk 1]
```diff
@@ -37,9 +37,14 @@ public bool CheckIntercept(string sender, string message)
             return "A scenario is already running.";
 
         string path = string.Join(' ', message.ToLower().Trim().Split()[1..]);
-        executingTask = Task.Run(async () => await ExecuteFileAsync(server, path));
+        if (!File.Exists(path))
+            return $"File {path} not found.";
+        else
+        {
+            executingTask = Task.Run(async () => await ExecuteFileAsync(server, path));
```
Comment (wessupermare) [inline Plugins/ScenarioLoader/Plugin.cs:44]:
```suggestion
            executingTask = ExecuteFileAsync(server, path);
```

This'll help with exception propagation.

File: Plugins/ScenarioLoader/Plugin.cs [hunk 2]
```diff
@@ -55,29 +60,67 @@ private async Task ExecuteFileAsync(IServer server, string path)
...
+                        if (parts.Length > 11)
+                        {
+                            string joined = string.Join(' ', parts);
+
+                            string[] split_flight = joined.Split(';');
+
+                            foreach (string str in split_flight) {
+
+                                Console.WriteLine(str + '\n');
+                            }
```
Comment (wessupermare) [inline Plugins/ScenarioLoader/Plugin.cs:73]:
```suggestion
                            foreach (string str in split_flight)
                                Console.WriteLine(str + '\n');
```
For style consistency, either remove the braces or put the opening brace on its own line.

File: Plugins/ScenarioLoader/Plugin.cs [hunk 3]
```diff
@@ -55,29 +60,67 @@ private async Task ExecuteFileAsync(IServer server, string path)
...
+                            string[] spawn_data = split_flight[0].Split(' ');
+                            string flightdata_joined = split_flight[1];
+                            string remarks = split_flight[2];
+                            string route = split_flight[3];
+
+                            string[] flightdata = flightdata_joined.Split(' ');
+
+                            IAircraft? aircraft = server.SpawnAircraft(
+                                                        parts[1],
+                                                        new(char.Parse(flightdata[0]), char.Parse(flightdata[1]), flightdata[2], flightdata[3], flightdata[4], new(), new(), flightdata[7], flightdata[8], 0, 0, 0, 0, "NONE", remarks, route),
```
Comment (wessupermare) [inline Plugins/ScenarioLoader/Plugin.cs:84]:
```suggestion
                                                        new(flightdata[0].Single(), flightdata[1].Single(), flightdata[2], flightdata[3], flightdata[4], new(), new(), flightdata[7], flightdata[8], 0, 0, 0, 0, "NONE", remarks, route),
```

`char.Parse` isn't safe. Consider either `.Single()` if you're confident in the data or either `[0]` or `char.TryParse()` if you're less sure.

**Review by wessupermare — CHANGES_REQUESTED** [review]
Good stuff generally speaking. Just a couple style & input safety things to check into then you should be all good.

File: Plugins/ScenarioLoader/Plugin.cs (hunk 1 above)
Comment (NikoDervaric) [reply Plugins/ScenarioLoader/Plugin.cs:44]: Change accepted for better exception propagation.

File: Plugins/ScenarioLoader/Plugin.cs (hunk 2 above)
Comment (NikoDervaric) [reply Plugins/ScenarioLoader/Plugin.cs:73]: Removed braces for consistency.

File: Plugins/ScenarioLoader/Plugin.cs (hunk 3 above)
Comment (NikoDervaric) [reply Plugins/ScenarioLoader/Plugin.cs]: Switched to `Single()`

**Comment by NikoDervaric** [comment]
All changes applied as per suggestions.

File: Plugins/ScenarioLoader/Plugin.cs [hunk 4]
```diff
@@ -55,29 +60,65 @@ private async Task ExecuteFileAsync(IServer server, string path)
...
+                            string[] spawn_data = split_flight[0].Split(' ');
+                            string flightdata_joined = split_flight[1];
+                            string remarks = split_flight[2];
+                            string route = split_flight[3];
+
+                            string[] flightdata = flightdata_joined.Split(' ');
+
+                            IAircraft? aircraft = server.SpawnAircraft(
+                                                        parts[1],
+                                                        new(flightdata[0].Single(), flightdata[1].Single(), flightdata[2], flightdata[3], flightdata[4], new(), new(), flightdata[7], flightdata[8], 0, 0, 0, 0, "NONE", remarks, route),
```
Comment (wessupermare) [inline Plugins/ScenarioLoader/Plugin.cs:82]: These calls to `Single()` need a check before they are safe. Consider adding a check to ensure that `flightdata[0].Length == 1 && flightdata[1].Length == 1` before line 80.

**Review by wessupermare — CHANGES_REQUESTED** [review]
Almost there. As discussed over Discord, just putting this here as a reminder. :)

**Review by wessupermare — APPROVED** [review]

**Pull request merged**